community/codegen/src/main/java/org/neo4j/codegen/source/SourceCode.java

```diff
@@ -61,6 +63,24 @@ public String toString()
         {
             return "SourceCode";
         }
+
+        private String fullClasspath()
+        {
+            StringBuilder classpath = new StringBuilder( System.getProperty( "java.class.path" ) );
+
+            ClassLoader classLoader = SourceCode.class.getClassLoader();
+            if ( classLoader instanceof URLClassLoader )
+            {
+                URL[] classpathUrls = ((URLClassLoader) classLoader).getURLs();
+                for ( URL classpathUrl : classpathUrls )
+                {
+                    classpath.append( File.pathSeparator );
+                    classpath.append( classpathUrl.getPath() );
+                }
+            }
```

Comment:
If we are extracting the classpath from the ClassLoader, perhaps we should extract it from the ClassLoader that the user passed in, and perhaps we should go through the entire ClassLoader hierarchy and not just the first ClassLoader.